qutebrowser/mainwindow/prompt.py

```diff
@@ -709,10 +727,26 @@ def item_focus(self, which):
         if not idx.isValid():
             idx = last_index if which == 'prev' else first_index
 
+        idx = self._do_completion(idx, which)
+
         selmodel.setCurrentIndex(
             idx, QItemSelectionModel.ClearAndSelect | QItemSelectionModel.Rows)
         self._insert_path(idx, clicked=False)
 
+    def _do_completion(self, idx, which):
+        while not self._file_model.fileName(idx).startswith(self._to_complete):
+            if which == 'prev':
+                idx = self._file_view.indexAbove(idx)
+            else:
+                assert which == 'next', which
+                idx = self._file_view.indexBelow(idx)
+
+            # wrap around if we arrived at beginning/end
+            if not idx.isValid():
+                idx = QModelIndex()
```

Comment:
I think the thing you get from `indexBelow` if there's no valid index anymore will already be `QModelIndex()`

Reply:
I'll have a look into that later